packages/grafana-ui/src/components/uPlot/config/UPlotScaleBuilder.ts

```diff
@@ -156,11 +156,19 @@ export class UPlotScaleBuilder extends PlotConfigBuilder<ScaleProps, Scale> {
 
       // if all we got were hard limits, treat them as static min/max
       if (hardMinOnly) {
-        minMax[0] = hardMin!;
+        if (scale.distr === 3 && hardMin! <= 0) {
+          // guard log scales from <= 0 user-defined limits
```

Comment:
Did you want to do anything here? or should the condition reversed?

Reply:
i can reverse it, but it doesnt read as easily, imo:

```js
if (scale.distr !== 3 || hardMin! > 0) {
  minMax[0] = hardMin!;
}
```

i think JS minifiers will handle inverting it anyways.